### ISSUE TITLE
Nextjs Build Failed with type: 'WorkerError'

### ISSUE THREAD

**Issue by m3rashid** [vercel/next.js#38324]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

  Operating System:
      Platform: linux
      Arch: x64
      Version: #43-Ubuntu SMP Wed Jun 15 12:54:21 UTC 2022
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: 1.22.18
      pnpm: N/A
    Relevant packages:
      next: 12.2.1-canary.2
      eslint-config-next: 12.1.6
      react: 18.2.0
      react-dom: 18.2.0

### What browser are you using? (if relevant)

Using Brave (But it is irrelevent imo because error in building, dev works fine)

### How are you deploying your application? (if relevant)

not deployed yet

### Describe the Bug

Build fails with the following error messages on the console

```js
yarn run v1.22.18
$ next build
info  - Loaded env from /home/rashid/code/projects/blog-web-next/.env
info  - Linting and checking validity of types  
info  - Creating an optimized production build  
info  - Compiled successfully
info  - Collecting page data ..node:internal/child_process/serialization:127
    const string = JSONStringify(message) + '\n';
```

{ the below error repeats 7-8 times }
```js
TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'ClientRequest'
    |     property 'socket' -> object with constructor 'TLSSocket'
    --- property '_httpMessage' closes the circle
    at stringify (<anonymous>)
    at writeChannelMessage (node:internal/child_process/serialization:127:20)
    at process.target._send (node:internal/child_process:839:17)
    at process.target.send (node:internal/child_process:739:19)
    at reportError (/home/rashid/code/projects/blog-web-next/node_modules/next/dist/compiled/jest-worker/processChild.js:1:1995)
    at reportClientError (/home/rashid/code/projects/blog-web-next/node_modules/next/dist/compiled/jest-worker/processChild.js:1:1671)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
info  - Collecting page data .node:internal/child_process/serialization:127
    const string = JSONStringify(message) + '\n';
```

{ and then finally this error }
```js
> Build error occurred
Error: Call retries were exceeded
    at ChildProcessWorker.initialize (/home/rashid/code/projects/blog-web-next/node_modules/next/dist/compiled/jest-worker/index.js:1:11661)
    at ChildProcessWorker._onExit (/home/rashid/code/projects/blog-web-next/node_modules/next/dist/compiled/jest-worker/index.js:1:12599)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess.emit (node:domain:475:12)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  type: 'WorkerError'
```

### Expected Behavior

Build create Successfully

### Link to reproduction

https://github.com/m3rashid/blog-web-next

### To Reproduce

run the following command

```bash
yarn build
```

**Comment by m3rashid** [comment]
I have checked in the next@canary version also,
Same error persists in canary version also
Wrote this in the initial issue also

**Comment by Cahmoraes** [comment]
I'm having the same problem.

Link to reproduction:
https://github.com/Cahmoraes/podcastr

**Comment by asavienko** [comment]
Have the same problem

**Comment by omeersari** [comment]
Same error, but in my local when I `yarn build`, it works, however when I try to deploy my app in vercel, it gives me this error.

`"next": "12.2.0"`

**Comment by balazsorban44** [comment]
I traced it back to some `axios`-related issues. 

Given this code in `pages/[slug].js`

```js
import axios from "axios"

export default function Page() {}

export async function getStaticProps() {
  return { props: {} }
}

export async function getStaticPaths() {
  try {
    await axios.post("https://cubicle.vercel.app/api/post/card")
  } catch (error) {
    console.log(error)
  } finally {
    return { paths: [], fallback: true }
  }
}
```

The error appears.

**Comment by omeersari** [comment]
any solution to this ?

**Comment by ryanmr** [comment]
Hi all,

I recently encountered this with a project on `yarn build` with next latest (which at this [12.2.2](https://www.npmjs.com/package/next/v/12.2.2)), just as [balazsorban44](https://github.com/vercel/next.js/issues/38324#issuecomment-1180380093) mentioned. It was occurring in my ci runner, and I realized someone else on the team was using `getStaticPaths` referencing an api route that wasn't available in ci. For some reason though, `getStaticPaths` wanted to serialize the error that was produced (which was stemming from axios not being to reach the api), and the serialization failed.
